app/views/aspects/_no_contacts_message.haml

```diff
@@ -3,10 +3,27 @@
 -#   the COPYRIGHT file.
 
 #no_contacts.empty_message
-  = t('.you_should_add_some_more_contacts')
-  %br
-  %br
-  = t('.try_adding_some_more_contacts')
+  %p.lead
+    = t('.you_should_add_some_more_contacts')
+
+  %p
+    != t('.try_adding_some_more_contacts',
+         search_link: link_to(t('.search_link_text'), search_path),
+         invite_link: link_to(t('.invite_link_text'),
+                                "invitationsModal",
+                                id: "invitations-link",
+                                data: {toggle: "modal"}))
+
+  = render "shared/modal",
+      path: new_user_invitation_path,
+      id: "invitationsModal",
+      title: t("invitations.new.invite_someone_to_join")
+
   - if AppConfig.settings.community_spotlight.enable?
     != t('.or_spotlight', :link => link_to(t(".community_spotlight") , community_spotlight_path))
 
+-# This will have to be extracte in a JS view
```

Comment:
You don't want the view for the modal, you want one for the page where the message is displayed so you can setup the hook there instead of with inline JS.